scripts/remind_draft_preregistrations.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from django.db import transaction
+from django.utils import timezone
+
+from framework.celery_tasks import app as celery_app
+from osf.models import DraftRegistration, MetaSchema, QueuedMail
+from osf.models.queued_mail import PREREG_REMINDER, PREREG_REMINDER_TYPE, queue_mail
+
+from website.app import init_app
+from website import settings
+
+from scripts.utils import add_file_logger
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def main(dry_run=True):
+
+    for prereg in find_neglected_prereg_within_reminder_limit():
+        if dry_run:
+            logger.warn('Dry run mode')
+        logger.info('Email of type prereg_reminder queued to send to {0}'.format(prereg.initiator.username))
+        if not dry_run:
+            with transaction.atomic():
+                queue_mail(
+                    to_addr=prereg.initiator.username,
+                    mail=PREREG_REMINDER,
+                    send_at=timezone.now(),
+                    user=prereg.initiator,
+                    fullname=prereg.initiator.fullname,
+                    prereg_url=prereg.absolute_url,
+                    draft_id=prereg._id,
+                )
+
+def find_neglected_prereg_within_reminder_limit():
+
+    queue_data = QueuedMail.objects.filter(
+        email_type=PREREG_REMINDER_TYPE,
+    ).values('data')
```

Comment:
Can this queue_data be limited based on time as well? Otherwise the size of this query (which is then modified in python) may get undesirably large.
Potential code below (untested!!!!!!!):

```
queue_data = QueuedMail.objects.filter(
    email_type=PREREG_REMINDER_TYPE,
    data__draft_id__isnull=False, # should remove entries with no draft_id
    send_at__gte=timezone.now() - settings.PREREG_AGE_LIMIT # I think this works, because it should only returned queued mails that could potentially be queued twice, but don't trust me.
).values_list('data', flat=True)

already_queued = [entry['draft_id'] for entry in queue_data]
```

I tried in vain to get the already_queued directly through the values_list but couldn't find a way to directly extract the data in the jsonfield :(

Reply:
@alexschiller, I originally limited it on time, and then changed it. I guess I thought it was weird that more things would queue that would never be sent. If that's not a problem, then I agree with you. Do you have a suggestion for a time limit. I had it just as two weeks at first (because that's the it's the PREREG_WAIT_TIME)... but that's kind of arbitrary.

Reply:
Update: you were right all along I just didn't pay enough attention reading your original comment.